createSession: "AssertionError: serverNonce instanceof Buffer" using userName and password

I'm trying to connect to a Server using a username and password, connecting anonymous works, but using a username and password results in the following error:
```
AssertionError: serverNonce instanceof Buffer
{ [AssertionError: serverNonce instanceof Buffer]
  name: 'AssertionError',
  actual: undefined,
  expected: undefined,
  operator: undefined,
  message: 'serverNonce instanceof Buffer',
  generatedMessage: false }
```

Sure,
I got node-opcua using npm, regarding to the package.json in the node_modules/node-opcua folder it is version 0.0.52.
The Server is a industrial SPS ( I think it is this one: http://www.br-automation.com/de/produkte/steuerungssysteme/x20-system/ ) with a build in OPC-UA Server. I do not have any further information about the kind of server, however connecting using other tools like UAExpert works fine.
Stack Trace:
AssertionError: serverNonce instanceof Buffer
    at createUserNameIdentityToken (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\opcua_client.js:309:5)
    at OPCUAClient.createUserIdentityToken (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\opcua_client.js:357:33)
    at OPCUAClient._activateSession (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\opcua_client.js:394:10)
    at D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\opcua_client.js:556:18
    at D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\opcua_client.js:235:9
    at modified_callback (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\client_secure_channel_layer.js:894:24)
    at process_request_callback (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\client_secure_channel_layer.js:72:5)
    at _on_message_received (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\client\client_secure_channel_layer.js:113:5)
    at emit (events.js:118:17)
    at MessageBuilder._decode_message_body (D:\Projekte\OPC-UA\NodeTest\node_modules\node-opcua\lib\misc\message_builder.js:445:22)

I am out of the office until 02.05.2016.

Während meiner Abwesenheit habe ich nur eingeschränkten Mail-Zugriff.
In dringenden Fällen wenden Sie sich bitte an Peter.Oltmanns@pol-it.de


Note: This is an automated response to your message  "Re:
[node-opcua/node-opcua] createSession: "AssertionError: serverNonce
instanceof Buffer" using userName and password (#196)" sent on 20.04.2016
08:35:09.

This is the only notification you will receive while this person is away.

I am out of the office until 02.05.2016.

Während meiner Abwesenheit habe ich nur eingeschränkten Mail-Zugriff.
In dringenden Fällen wenden Sie sich bitte an Peter.Oltmanns@pol-it.de


Note: This is an automated response to your message  "Re:
[node-opcua/node-opcua] createSession: "AssertionError: serverNonce
instanceof Buffer" using userName and password (#196)" sent on 25.04.2016
22:48:21.

This is the only notification you will receive while this person is away.

I am out of the office until 02.05.2016.

Während meiner Abwesenheit habe ich nur eingeschränkten Mail-Zugriff.
In dringenden Fällen wenden Sie sich bitte an Peter.Oltmanns@pol-it.de


Note: This is an automated response to your message  "Re:
[node-opcua/node-opcua] createSession: "AssertionError: serverNonce
instanceof Buffer" using userName and password (#196)" sent on 26.04.2016
10:48:03.

This is the only notification you will receive while this person is away.

I am out of the office until 02.05.2016.

Während meiner Abwesenheit habe ich nur eingeschränkten Mail-Zugriff.
In dringenden Fällen wenden Sie sich bitte an Peter.Oltmanns@pol-it.de


Note: This is an automated response to your message  "Re:
[node-opcua/node-opcua] createSession: "AssertionError: serverNonce
instanceof Buffer" using userName and password (#196)" sent on 26.04.2016
11:05:23.

This is the only notification you will receive while this person is away.

--- a/lib/client/opcua_client.js
+++ b/lib/client/opcua_client.js
@@ -306,12 +306,25 @@ function createUserNameIdentityToken(session, userName, password) {
     }
 
     var serverCertificate = session.serverCertificate;
+    // if server does not provide certificate use unencrypted password
+    if (serverCertificate == null) {
+        var userIdentityToken = new UserNameIdentityToken({
+	        userName: userName,
+            password: new Buffer(password, "utf-8"),
+            encryptionAlgorithm: null,
+            policyId: userTokenPolicy.policyId
+	    });
+       return userIdentityToken; 
+    }
+    
     assert(serverCertificate instanceof Buffer);
 
     serverCertificate = crypto_utils.toPem(serverCertificate, "CERTIFICATE");
     var publicKey = crypto_utils.extractPublicKeyFromCertificateSync(serverCertificate);
 
     var serverNonce = session.serverNonce;
+    // if serverNonce not specified by server
+	if (serverNonce == null) serverNonce = new Buffer(0);
     assert(serverNonce instanceof Buffer);
 
 

